src/devices/Tm1637/Tm1637.cs

```diff
@@ -0,0 +1,299 @@
+        /// bit 2 = c      f   b
+        /// bit 3 = d      |_g_|
+        /// bit 4 = e      |   |
+        /// bit 5 = f      e   c
+        /// bit 6 = g      |_d_|  .dp
+        /// bit 7 = dp
+        /// 
+        /// Representation of the number 0 so lighting segments a, b, c, d, e and F is then 0x3f
+        /// </summary>
+        /// <param name="rawData">The raw data array to display, size of the array has to be 6 maximum</param>
+        private void Display(ReadOnlySpan<byte> rawData)
+        {
+            if (rawData.Length > MaxSegments)
+                throw new ArgumentException($"Maximum number of segments for TM1637 is {MaxSegments}");
+
+            // Prepare the buffer with the right order to transfer
+            byte[] toTransfer = new byte[MaxSegments];
+
+            for (int i = 0; i < rawData.Length; i++)
+                toTransfer[_segmentOrder[i]] = rawData[i];
+
+            for (int j = rawData.Length; j < MaxSegments; j++)
+                toTransfer[_segmentOrder[j]] = (byte)Character.Nothing;
+            _lastDisplay = toTransfer;
+
+            StartTransmission();
+            // First command is set data
+            WriteByte((byte)DataCommand.DataCommandSetting);
+            StopTransmission();
+            StartTransmission();
+            // Second command is set address to automatic
+            WriteByte((byte)DataCommand.DataCommandSetting);
+            // Transfer the data
+            for (int i = 0; i < MaxSegments; i++)
+                WriteByte(toTransfer[i]);
+
+            StopTransmission();
+            StartTransmission();
+            // Set the display on/off and the brightness
+            WriteByte((byte)((ScreenOn ? DisplayCommand.DisplayOn : DisplayCommand.DisplayOff) + _brightness));
+            StopTransmission();
+        }
+
+        /// <summary>
+        /// Displays a series of prebuild characters including the dot or not
+        /// You can build your won characters with the primitives like Bottom, Top, Dot
+        /// </summary>
+        /// <param name="rawData">The Character to display</param>
+        public void Display(ReadOnlySpan<Character> rawData)
+        {
+            Display(MemoryMarshal.AsBytes(rawData));
+        }
+
+
+        /// <summary>
+        /// Displays a raw data at a specific segment address from 0 to 6
+        /// 
+        /// bit 0 = a       _a_
+        /// bit 1 = b      |   |
+        /// bit 2 = c      f   b
+        /// bit 3 = d      |_g_|
+        /// bit 4 = e      |   |
+        /// bit 5 = f      e   c
+        /// bit 6 = g      |_d_|  .dp
+        /// bit 7 = dp
+        /// 
+        /// Representation of the number 0 so lighting segments a, b, c, d, e and F is then 0x3f
+        /// </summary>
+        /// <param name="segmentAddress">The segment address from 0 to 6</param>
+        /// <param name="rawData">The rsegemnets to display</param>
+        public void Display(byte segmentAddress, Character rawData)
```

Comment:
Given that segmentAddress refers to a numbered position, I would change this so that it takes an int instead (and add validation so that the number is from 0 to 6) and change the name of the parameter to be something like segmentPosition or something. The remarks should also say that this segmentPosition starts at index 0.

Reply:
ok to rename but what is the advantage of changing the type from byte to int in this case? With byte, you're at least sure it starts at 0. And I'm anyway testing if we have more than the maximum number of 6 segments.